wdl-lint/src/rules/shellcheck.rs

```diff
@@ -396,6 +398,48 @@ fn to_bash_var(placeholder: &Placeholder, ty: Option<Type>) -> (String, bool) {
                     true,
                 );
             }
+            PrimitiveType::String => {
+                match placeholder.expr() {
+                    Expr::If(i) =>  {
+                        let (_, if_expr, else_expr) = i.exprs();
+                        if let (
+                            Expr::Literal(LiteralExpr::String(_)),
+                            Expr::Literal(LiteralExpr::String(_)),
+                        ) = (if_expr, else_expr)
+                        {
+                            return ("a".repeat(placeholder_len), true);
+                        }
+                    }
+                    Expr::Parenthesized(p) => {
+                        if let Expr::If(i) = p.expr() {
+                            let (_, if_expr, else_expr) = i.exprs();
+                            if let (
+                                Expr::Literal(LiteralExpr::String(_)),
+                                Expr::Literal(LiteralExpr::String(_)),
+                            ) = (if_expr, else_expr)
+                            {
+                                return ("a".repeat(placeholder_len), true);
+                            }
+                        }
+                    }
+                    Expr::Call(c) => {
+                        let target = c.target();
+                        if target.text() == "sep" {
+                            match c.arguments().nth(1) {
+                                Some(Expr::Call(c)) => {
+                                    if c.target().text() == "quote" || c.target().text() == "squote" {
+                                        // If the second argument is a quoted string, we can replace
+                                        // the placeholder with a literal.
+                                        return ("a".repeat(placeholder_len), true);
+                                    }
+                                }
+                                _ => {}
+                            }
+                        }
+                    }
```

Comment:
@a-frantz - I'm conflicted on this. It does solve some annoyances, but I'm not certain it won't have side effects. This treats `sep` calls where the second argument is wrapped in a `quote()` or `squote()` call as literal strings.

Reply:
I've pushed an update Arena.toml. I've been running it locally on the `workflows` repo and checking manually which lints get removed.

Reply:
I didn't review super close, but it looks like all that changed were "double quote" diagnostics? That suggests to me that this is safe(-ish at least)